stake-pool/js/src/schema.ts

```diff
@@ -0,0 +1,181 @@
+import {Schema, serialize, deserializeUnchecked} from 'borsh';
+import BN from 'bn.js';
+
+// Class wrapping a plain object
+export abstract class Assignable {
+  constructor(properties: {[key: string]: any}) {
+    Object.keys(properties).forEach((key: string) => {
+      this[key] = properties[key];
+    });
+  }
+
+  encode(): Buffer {
+    return Buffer.from(serialize(SCHEMA, this));
+  }
+
+  static decode<T extends Assignable>(data: Buffer): T {
+    return deserializeUnchecked(SCHEMA, this, data);
+  }
+}
+
+// Class representing a Rust-compatible enum, since enums are only strings or
+// numbers in pure JS
+export abstract class Enum extends Assignable {
+  enum: string;
+  constructor(properties: any) {
+    super(properties);
+    if (Object.keys(properties).length !== 1) {
+      throw new Error('Enum can only take single value');
+    }
+    this.enum = '';
+    Object.keys(properties).forEach(key => {
+      this.enum = key;
+    });
+  }
+}
+
+export class Fee extends Assignable {
+  denominator: BN;
+  numerator: BN;
+}
+
+export class AccountType extends Enum {}
+
+export class AccountTypeEnum extends Assignable {}
+
+export enum AccountTypeKind {
+  Uninitialized = 'Uninitialized',
+  StakePool = 'StakePool',
+  ValidatorList = 'ValidatorList',
+}
+
+export class StakePoolAccount extends Assignable {
+  accountType: AccountType;
+  manager: PublicKey;
+  staker: PublicKey;
+  depositAuthority: PublicKey;
+  withdrawBumpSeed: number;
+  validatorList: PublicKey;
+  reserveStake: PublicKey;
+  poolMint: PublicKey;
+  managerFeeAccount: PublicKey;
+  totalStakeLamports: BN;
+  poolTokenSupply: BN;
+  lastUpdateEpoch: BN;
+  fee: Fee;
+}
+
+export class ValidatorListAccount extends Assignable {
+  accountType: AccountType;
+  maxValidators: number;
+  validators: [ValidatorStakeInfo];
+}
+export class ValidatorStakeInfo extends Assignable {
+  status: StakeStatus;
+  voteAccountAddress: PublicKey;
+  stakeLamports: BN;
+  lastUpdateEpoch: BN;
+}
+export class StakeStatus extends Enum {}
+
+export class StakeStatusEnum extends Assignable {}
+
+export enum StakeStatusKind {
+  Active = 'Active',
+  DeactivatingTransient = 'DeactivatingTransient',
+  ReadyForRemoval = 'ReadyForRemoval',
+}
+
+export class PublicKey extends Assignable {
+  value: BN;
+}
+
+export const SCHEMA: Schema = constructStakePoolSchema();
```

Comment:
You can add to `SOLANA_SCHEMA` defined at https://github.com/solana-labs/solana/blob/ec1a307a7c2618b9bdea7b82a39a36ba194ae0d0/web3.js/src/util/borsh-schema.ts#L34